src/campaigns/campaigns.controller.ts

```diff
@@ -277,6 +279,44 @@ export class CampaignsController {
         ...raceTargetDetails,
         electionType: L2DistrictType,
         electionLocation: L2DistrictName,
+        districtManuallySet: true,
+      },
+    })
+  }
+
+  @Put('mine/race-target-details')
+  @UseCampaign()
+  async updateRaceTargetDetails(@ReqCampaign() campaign: Campaign) {
+    if (!campaign?.details?.positionId || !campaign.details.electionDate) {
+      throw new BadRequestException(
+        `Error: The campaign has no ballotready 'positionId' or electionDate and likely hasn't selected an office yet`,
+      )
+    }
+    const raceTargetDetails =
+      await this.elections.getBallotReadyMatchedRaceTargetDetails(
+        campaign.details.positionId,
+        campaign.details.electionDate,
+      )
+    if (!raceTargetDetails) {
+      throw new NotFoundException(
+        'Failed to fetch the raceTargetDetails for the requested campaign',
+      )
+    }
+    const { district, winNumber, voterContactGoal, projectedTurnout } =
+      raceTargetDetails
+    const { L2DistrictType, L2DistrictName } = district
+    return this.campaigns.updateJsonFields(campaign.id, {
+      pathToVictory: {
+        districtId: district.id,
+        electionType: L2DistrictType,
+        electionLocation: L2DistrictName,
+        winNumber,
+        voterContactGoal,
+        projectedTurnout,
+        source: P2VSource.ElectionApi,
+        p2vStatus: P2VStatus.complete,
+        p2vCompleteDate: new Date().toISOString().slice(0, 10),
```

Comment:
Looks like we have 2 _different_ ways that we format this date:

https://github.com/thegoodparty/gp-api/blob/14ca0eb28337579ea0ad8da4543f99c3bf4af7bf/src/admin/services/adminP2V.service.ts#L19
https://github.com/thegoodparty/gp-api/blob/bada25845757d30fa60aa9a9210e880d2afdde5f/src/pathToVictory/services/pathToVictory.service.ts#L410

Might be worth taking a look into whether or not we want to reconcile this change w/ these other two places where this value is set? 🤔